src/app/filter_utils.ts

```diff
@@ -145,6 +145,14 @@ export function buildDateFilterForNDays(now: Date, days: number): DateFilter {
   };
 }
 
+export function buildDateFilterForNDaysFromNow(days: number): DateFilter {
+  const now = new Date().getTime();
+  return {
+    startDateTimeMs: now - days * MILLISECONDS_PER_DAY,
+    endDateTimeMs: now,
+  }
+}
+
```

Comment:
Not sure I follow this: If we only move back `x` days from _now_ (**16:14**), it means we'll only pull tweets sent from **16:14** `x` days from _now_.

What we need is to pull tweets sent from _midnight_ `x` days from now. e.g. if `x` is **7**, we need to pull all tweets from _midnight_, last Monday.

Reply:
@kilemensi The twitter api definition if last days is a little bit different. If for example we need tweets from the last 7 days, it will only allow us to pull tweets from last Monday at the current time (16.22), not from midnight.

Reply:
The difference should be precise to the current time

Reply:
Cool, can you link to the specific docs so we have it documented @kelvinkipruto ?

Reply:
@kilemensi It is not specified in the docs, but it was the error message returned.

Reply:
Sure, then just add a comment there @kelvinkipruto. Can't we use the same method rather than create a new one?

Reply:
@kilemensi An overview of some other methods using the original method shows that they wouldn't be affected much. For us to re use the original method `buildDateFilterForNDays`, I think we can add another parameter to it, say `precise` which is false by default, then use it to calculate the date precisely.

Reply:
That's the kind of of thinking I love to see @kelvinkipruto ... or even an enum with now & midnight options. This is TypeScript after all.

Reply:
@kilemensi I have used an enum. It works perfect